File: sherlock/pascal/sherlock.pas
function IsValid(s: string): Boolean;
var
  i, j, originI, cpt, previousCmpt: Integer;
  resultValue: Boolean;
  c: Char;
begin
  i := 1;
  j := 1;
  originI := 0;
  cpt := 0;
  previousCmpt := 0;
  resultValue := True;
  
  while i <= Length(s) do
  begin
    cpt := 0;
    c := s[i];
    originI := i;
  
    while j <= Length(s) do
    begin
      if (s[j] = c) and (j <> originI) then
      begin
        cpt := cpt + 1;
      end;
      j := j + 1;
    end;
    j := 1;
  
    if cpt > 1 then
    begin
      resultValue := False;
      Break;
    end
    else
    begin
      if i = 1 then
      begin
        previousCmpt := cpt;
      end
      else
      begin
        if cpt <= previousCmpt then
        begin
          //true 
          resultValue := True;
        end
        else
        begin
          resultValue := False;
        end;
      end;
    end;
    i := i + 1;
  end;
  IsValid:=resultValue;
end;

begin
  WriteLn(IsValid('lhcceexxnxzy'));
end.

File: sherlock/python/sherlock.py
def is_valid(s):
    i = 0
    j = 0
    origin_i = 0
    cpt = 0
    previous_cpt = 0
    result = True
    c = ''
  
    while i < len(s):
        cpt = 0
        c = s[i]
        origin_i = i
  
        while j < len(s):
            if s[j] == c and j != origin_i:
                cpt += 1
            j += 1
        j = 0
  
        if cpt > 1:
            result = False
            break
        else:
            if i == 0:
                previous_cpt = cpt
            else:
                if cpt <= previous_cpt:
                    result = True
                else:
                    result = False
            previous_cpt = cpt
        i += 1
    return result
  
print(is_valid('lahcenxzy'))